Cockpit 306 (#682)
cockpit-project/cockpit-project.github.io: _posts/2023-11-29-cockpit-306.md

---
title: Cockpit 306
author: pitti
date: '2023-11-29'
tags: cockpit, machines, podman, ostree
slug: cockpit-306
category: release
summary: 'kdump automation, OSTree redesign, deleting intermediate podman images'
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 306, cockpit-machines 303, cockpit-podman 82, and cockpit-ostree 198.1:

## Kdump: Add Ansible/shell automation

The Kdump page can generate an [Ansible](https://www.ansible.com/automation) role or a shell script for replicating the current kdump configuration to other machines.

> ![screenshot of add ansible/shell automation]({{ site.baseurl }}/images/306-kdump-add-ansible-shell-automation.png)

Reviewer: ```suggestion
![screenshot of the kdump page with an Ansible button]({{ site.baseurl }}/images/306-kdump-add-ansible-shell-automation.png)
```

> ![screenshot of add ansible/shell automation]({{ site.baseurl }}/images/306-kdump-add-ansible-shell-automation-2.png)

Reviewer: ```suggestion
![screenshot of shell automation dialog with the "Ansible" tab selected and the "shell script" tab unselected]({{ site.baseurl }}/images/306-kdump-add-ansible-shell-automation-2.png)
```

> ## OSTree: Redesign with new features: status, clean up, reset, and pin

Author: This is too long, it's a bit ugly when the title wraps to two lines. @garrett: Perhaps just "OSTree: Redesign with new features" and we leave the details to the paragraphs?

Reviewer: Sure!

```suggestion
## OSTree: Redesign, with new features
```

The OSTree page for software updates has been redesigned and includes several new features.

> Update status is now listed in the "Status" card, and details about the current OSTree repository and branch are visible in the "OSTree source" card:

Reviewer: ```suggestion
Update status is now listed in the "Status" card, and details about the current OSTree repository and branch are visible in the "OSTree source" card.
```

> ![screenshot of status, clean up, reset, and pin]({{ site.baseurl }}/images/306-ostree-redesign-with-new-features-status-clean-up-reset-and-pin.png)

Reviewer: ```suggestion
![new design of the OSTree page, showing off the status card]({{ site.baseurl }}/images/306-ostree-redesign-with-new-features-status-clean-up-reset-and-pin.png)
```

Unused deployments and package cache can be removed in using the "clean up" action.

> ![screenshot of status, clean up, reset, and pin]({{ site.baseurl }}/images/306-ostree-redesign-with-new-features-status-clean-up-reset-and-pin-2.png)

Reviewer: ```suggestion

![screenshot of the clean up modal dialog, with cache selected and deployments unselected by default]({{ site.baseurl }}/images/306-ostree-redesign-with-new-features-status-clean-up-reset-and-pin-2.png)
```

Alt text and blank separator line between text and image

A new "Reset" feature has been added, which can remove layered and overriden packages.

![reset modal dialog]({{ site.baseurl }}/images/306-ostree-redesign-with-new-features-status-clean-up-reset-and-pin-3.png)

Deployments can be pinned to persist even when new deployments trigger an auto-cleanup and unpinned.

![screenshot showing a pinned deployment]({{ site.baseurl }}/images/306-ostree-redesign-with-new-features-status-clean-up-reset-and-pin-4.png)
## Machines: Change "Add disk" default behavior

The "Always attach" Persistent option will now be set by default.

## Podman: Delete intermediate images

> Intermediate images have no tags or other identifiers, they are usually intermediate layers from building container images, or leftovers from moved tags. These can now be deleted:

Reviewer: ```suggestion
Intermediate images have no tags or other identifiers. They are usually intermediate layers from building container images or leftovers from moved tags. Intermediate images can now be deleted within Cockpit Podman.
```

> ![screenshot of delete intermediate images]({{ site.baseurl }}/images/306-podman-delete-intermediate-images.png)

Reviewer: ```suggestion
![screenshot of the "delete intermediate images" modal dialog]({{ site.baseurl }}/images/306-podman-delete-intermediate-images.png)
```
